inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1417,6 +1417,7 @@ void GNAGraphCompiler::PermutePrimitive(InferenceEngine::CNNLayerPtr layer) {
     }
     auto layerOrder = layer->GetParamAsInts("order");
     auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+    IE_ASSERT(!layer->insData.empty());
```

Comment:
i would use throw_gna_layer_exception, macro in compiler code, so it is easier to track issues related to certain layer

Reply:
I did that in the version of PR (#903) targeted to master branch. However, we have many pieces of code like this. I propose to create a new assert-like macro for such empty pointers corresponding to layers.

Reply:
Ok for now we could use just something like IE_ASSERT(expr) << LAYER_NAME(layer),  or create GNA_LAYER_ASSERT(expr, layer) which effectively transformed to previous code

Reply:
Yes, but let's leave it this way in the release branch if you are not against that.